db/migrations/20160922213611_organization_isolation_segments.rb

```diff
@@ -0,0 +1,31 @@
+Sequel.migration do
+  up do
+    create_table :organizations_isolation_segments do
+      String :organization_guid, null: false
+      String :isolation_segment_guid, null: false
+
+      foreign_key [:organization_guid], :organizations, key: :guid, name: :fk_organization_guid
+      foreign_key [:isolation_segment_guid], :isolation_segments, key: :guid, name: :fk_isolation_segments_guid
+      primary_key [:organization_guid, :isolation_segment_guid], name: :organizations_isolation_segments_pk
+    end
+
+    collate_opts = {}
+    if self.class.name.match(/mysql/i)
+      collate_opts[:collate] = :utf8_bin
+    end
+
+    alter_table :organizations do
+      add_column :default_isolation_segment_guid, String, collate_opts
+
+      add_foreign_key [:guid, :default_isolation_segment_guid], :organizations_isolation_segments, :name => 'organizations_isolation_segments_pk'
+    end
+  end
+
+  down do
+    alter_table :organizations do
```

Comment:
Do you also need to remove the `default_isolation_segment_guid` column? Or does Sequel magically do that when dropping the foreign key?